### PR TITLE
Fix regression introduced for Eject Button

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Regression introduced in #3669

Target to upcoming because the recently introduced `enduredDamage` Battle Struct field fixes a small bug that applies here as well 

Other:

Fixes: #3633

### CODE REVIEW

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -5853,7 +5853,8 @@ static void Cmd_moveend(void)
                     if (IsBattlerAlive(battler)
                       && gBattlerAttacker != battler
                       && GetBattlerHoldEffect(battler, TRUE) == HOLD_EFFECT_EJECT_BUTTON
-                      && TARGET_TURN_DAMAGED
```
Comment (ghoulslash) [inline src/battle_script_commands.c:5856]: maybe better to have a BATTLER_TURN_DAMAGED macro thats the same as TARGET_TURN_DAMAGED but for a given battler id

Comment (AlexOn1ine) [reply src/battle_script_commands.c:5856]: Seems like it already exists `BATTLER_DAMAGED` but is only used once. Should I rename it? I like `BATTLER_TURN_DAMAGED` more

Comment (AlexOn1ine) [reply src/battle_script_commands.c:5856]: Went ahead and made the change.

**Comment by AlexOn1ine** [comment]
Also Renamed `jumpifaffectedbyprotect` to unused5. Fixes: #3633

File: src/battle_script_commands.c (hunk 1 above)
Comment (ghoulslash) [reply src/battle_script_commands.c:5856]: Yeah BATTLER_DAMAGED is fine

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**